core/trino-main/src/test/java/io/trino/execution/scheduler/TestStageTaskSourceFactory.java

```diff
@@ -593,15 +595,16 @@ private static HashDistributionTaskSource createHashDistributionTaskSource(
                 Optional.of(CATALOG),
                 targetPartitionSplitWeight,
                 targetPartitionSourceSize,
-                DataSize.of(4, GIGABYTE));
+                DataSize.of(4, GIGABYTE),
+                directExecutor());
     }
 
     @Test
     public void testSourceDistributionTaskSource()
     {
         TaskSource taskSource = createSourceDistributionTaskSource(ImmutableList.of(), ImmutableListMultimap.of(), 2, 0, 3 * STANDARD_WEIGHT, 1000);
         assertFalse(taskSource.isFinished());
-        assertEquals(taskSource.getMoreTasks(), ImmutableList.of());
+        assertEquals(getFutureValue(taskSource.getMoreTasks()), ImmutableList.of());
```

Comment:
As an afterthought. It would be great if we could cover the change with additional tests. For example in `TestStageTaskSourceFactory` we could create a testing `SplitSource` that doesn't load splits right away, but with some delay (either fixed or explicitly controlled by a settable future) and exercise both task sources that now can load tasks asynchronously based on the split loading.